$NetBSD: patch-ae,v 1.4 2026/03/29 07:29:05 taca Exp $

* Try load from PKG_SYSCONFDIR/dot.rd2rc.
* Depend on $0 strictly.

--- bin/rd2.orig	2026-03-29 07:08:13.265719334 +0000
+++ bin/rd2
@@ -78,6 +78,7 @@ class RD2Command
     @output_index = false
     @out_code = nil
     @from_rdo = false
+    @sysconf = "@PKG_SYSCONFDIR@/dot.rd2rc"
   end
 
   def run
@@ -173,7 +174,7 @@ class RD2Command
     if File.readable?(user_config)
       load user_config
     else
-      load "rd/dot.rd2rc"
+      load @sysconf
     end
   rescue LoadError, StandardError
     load "rd/dot.rd2rc"
@@ -181,7 +182,7 @@ class RD2Command
 
   def require_implicit_format_lib
     base = File.basename(@program_name, ".*").downcase
-    return unless /rd2.+/ =~ base
+    return unless /rd2[1-9][0-9]+/ =~ base
 
     load_format_library("rd/#{$&}-lib.rb")
   end
